lib/core-integration/src/Test/Integration/Framework/DSL.hs

```diff
@@ -2763,6 +2819,37 @@ updateWalletPassphraseViaCLI ctx wid ppOld ppNew ppNewConfirm = do
             err <- TIO.hGetContents stderr
             pure (c, out, err)
 
+updateWalletPassphraseWithMnemonicViaCLI
+    :: forall s m. (HasType (Port "wallet") s, MonadIO m)
+    => s
+    -> String
+        -- ^ Wallet id
+    -> [Text]
+        -- ^ Mnemonic
+    -> String
+        -- ^ New passphrase
+    -> String
+        -- ^ New passphrase (repeated for confirmation)
+    -> m (ExitCode, Text, Text)
+updateWalletPassphraseWithMnemonicViaCLI ctx wid mnemonic ppNew ppNewConfirm = do
+    let process = proc' commandName
+            [ "wallet", "update", "passphrase"
+            , "--port", show (ctx ^. typed @(Port "wallet"))
+            , "--mnemonic"
+            , wid
+            ]
+    liftIO $ withCreateProcess process $
+        \(Just stdin) (Just stdout) (Just stderr) h -> do
+            hPutStr stdin (T.unpack (T.unwords mnemonic) <> "\n\n")
+            hPutStr stdin (ppNew <> "\n")
+            hPutStr stdin (ppNewConfirm <> "\n")
+            hFlush stdin
```

Comment:
Computation [hClose](https://hackage.haskell.org/package/base-4.16.1.0/docs/System-IO.html#v:hClose) hdl makes handle hdl closed. Before the computation finishes, if hdl is writable its buffer is flushed as for [hFlush](https://hackage.haskell.org/package/base-4.16.1.0/docs/System-IO.html#v:hFlush).
```suggestion
```